srml/staking/src/lib.rs

```diff
@@ -16,7 +16,157 @@
 // You should have received a copy of the GNU General Public License
 // along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
 
-//! Staking manager: Periodically determines the best set of validators.
+//! # Staking Module
+//! 
+//! <!-- Original author of paragraph: @gavofyork -->
+//! The staking module is the means by which a set of network maintainers (known as "authorities" in some contexts and "validators" in others) are chosen based upon those who voluntarily place funds under deposit. Under deposit, those funds are rewarded under normal operation but are held at pain of "slash" (expropriation) should they be found not to bee discharging their duties properly. 
+//! 
+//! ## Overview 
+//! 
+//! ### Terminology
+//! <!-- Original author of paragraph: @gavofyork -->
+//! 
+//! - Staking: The process of locking up funds for some time, placing them at risk of slashing (loss) in order to become a rewarded maintainer of the network.
+//! - Validating: The process of running a node to actively maintain the network, either by producing blocks or guaranteeing finality of the chain.
+//! - Nominating: The process of placing staked funds behind one or more validators in order to share in any reward, and punishment, they take.
+//! - Stash account: The account holding an owner's funds used for staking.
+//! - Controller account: The account which controls am owner's funds for staking.
+//! - Era: A (whole) number of sessions which is the period that the validator set (and each validator's active nominator set) is recalculated and where rewards are paid out.
```

Comment:
English nitpick. From this sentence, it's not clear if "the period that the validator set is recalculated" is the definition of a session or the definition of an era. 

* "A (whole) number of sessions**,** which is the period" --> defining sessions
* "A (whole) number of sessions **that** is the period" --> defining era

Reply:
^MD meant to emphasize the comma after "sessions"

Reply:
great point.